src/common/buffer.cc

```diff
@@ -1321,9 +1310,11 @@ static ceph::spinlock debug_lock;
 
   void buffer::list::reserve(size_t prealloc)
   {
-    if (append_buffer.unused_tail_length() < prealloc) {
-      append_buffer = buffer::create_page_aligned(prealloc);
-      append_buffer.set_length(0);   // unused, so far.
+    if (_buffers.empty() || _buffers.back().unused_tail_length() < prealloc) {
```

Comment:
why not just put
```c++
if (get_append_buffer_unused_tail_length() < prealloc) {
  // ...
}
```

Reply:
I've missed that opportunity. Thanks, @tchaikov!